tests/test_client.py

```diff
@@ -1688,6 +1693,62 @@ def on_error(event):
             FeatureFlag('e')
         ]
 
+    def test_in_flight_event_request_tracking_in_notify(self):
+        delivery = QueueingDelivery()
+        configuration = Configuration()
+        configuration.configure(delivery=delivery, api_key='abc')
+
+        client = Client(configuration)
+        assert not client._request_tracker.has_in_flight_requests()
+
+        client.notify(Exception('Oh no'))
+        assert client._request_tracker.has_in_flight_requests()
+
+        delivery.flush_request_queue()
```

Comment:
I'm assuming this is synchronous and won't introduce a race condition with the following check?

Reply:
Yeah, [`QueueingDelivery` came from #375](https://github.com/bugsnag/bugsnag-python/pull/375/files#diff-9e08e8c7769db7b58089fd7659d75ca96df40bb7b5d50299d3a30abd54da9ad6R182-R193) and is entirely synchronous